File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.java
class Solution {
    public int[] solution(int[] arr, int[][] intervals) {
        int a1 = intervals[0][0];
        int b1 = intervals[0][1];
        int a2 = intervals[1][0];
        int b2 = intervals[1][1];
        
        int index = ((b1-a1)+1) + ((b2-a2)+1);

        int[] answer = new int[index];
        
        int start1 = 0;
        int start2 = (b1-a1)+1;
        
        
        while(true){
            if(a1 <= b1) answer[start1++] = arr[a1++];
            if(a2 <= b2) answer[start2++] = arr[a2++];
            if(a1 > b1 && a2 > b2) break;
        }
        
        return answer;
    }
}

File: C/프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int arr[], size_t arr_len, int** intervals, size_t intervals_rows, size_t intervals_cols) {
    int len = (intervals[0][1] - intervals[0][0] + 1) + (intervals[1][1] - intervals[1][0] + 1);
    int* answer = (int*)malloc(sizeof(int) * len);
    int a, b;
    int idx = 0;
    
    for(int i = 0; i < intervals_rows; i++){
        a = intervals[i][0];
        b = intervals[i][1];
        for(; a <= b; a++){
            answer[idx++] = arr[a];
        }
    }
    
    return answer;
}